modules/bibdk_favourite/includes/bibdk_favourite.forms.inc

```diff
@@ -84,15 +84,19 @@ function bibdk_favourite_check_for_new_agency() {
           unset($_SESSION['bibdk_favourites']);
         }
         // get the agency
-        $agency = new TingAgency($_GET['agency']);
+        try {
+          $agency = vip_core_findlibrary($_GET['agency']);
+        } catch (Exception $e) {
+          drupal_set_message($e->getMessage(), 'error');
+        }
         global $language;
-        $name = $agency->getBranch()->getBranchName($language->language);
+        $name = $agency->getBranchName();
```

Comment:
Bare lige for at være helt sikker - sproghåndtering foregår i VIP Core modulet, ikke?